pkg/proc/types.go

```diff
@@ -365,35 +367,75 @@ func (bi *BinaryInfo) loadDebugInfoMaps(debugLineBytes []byte, wg *sync.WaitGrou
 
 		case dwarf.TagSubprogram:
 			ok1 := false
+			inlined := false
 			var lowpc, highpc uint64
+			if inval, ok := entry.Val(dwarf.AttrInline).(int64); ok {
+				inlined = inval == 1
+			}
 			if ranges, _ := bi.dwarf.Ranges(entry); len(ranges) == 1 {
 				ok1 = true
 				lowpc = ranges[0][0]
 				highpc = ranges[0][1]
 			}
 			name, ok2 := entry.Val(dwarf.AttrName).(string)
-			if ok1 && ok2 {
+			var fn Function
+			if (ok1 == !inlined) && ok2 {
+				if inlined {
+					abstractOriginNameTable[entry.Offset] = name
+				}
 				if pu != nil {
-					pu.functions = append(pu.functions, Function{
+					fn = Function{
 						Name:  name,
 						Entry: lowpc, End: highpc,
 						offset: entry.Offset,
 						cu:     &compileUnit{},
-					})
+					}
+					pu.functions = append(pu.functions, fn)
 				} else {
 					if !cu.isgo {
 						name = "C." + name
 					}
-					bi.Functions = append(bi.Functions, Function{
+					fn = Function{
 						Name:  name,
 						Entry: lowpc, End: highpc,
 						offset: entry.Offset,
 						cu:     cu,
-					})
+					}
+					bi.Functions = append(bi.Functions, fn)
+				}
+			}
+			if entry.Children {
+				for {
+					entry, err = reader.Next()
+					if err != nil {
+						break outer
+					}
+					if entry.Tag == 0 {
+						break
+					}
+					if entry.Tag == dwarf.TagInlinedSubroutine {
+						originOffset := entry.Val(dwarf.AttrAbstractOrigin).(dwarf.Offset)
+						name := abstractOriginNameTable[originOffset]
```

Comment:
I'm worried that an inlined function will appear before its abstract entry and this won't find anything. I don't think there's anything preventing this in the standard.

Reply:
I understand that concern but it seems unlikely. The concrete instance refers to the offset of it's abstract root, meaning that offset must already be known. Seems inefficient to make another pass to fill in the offset after if the abstract root is written later.